hcctl/src/main.rs

```diff
@@ -146,3 +140,41 @@ fn list(settings: Settings) -> Result<()> {
 
     Ok(())
 }
+
+fn human_readable_duration(now: &DateTime<Utc>, date_str: &str) -> Result<String> {
+    let date = DateTime::parse_from_rfc3339(&date_str)?;
+    let duration = now.signed_duration_since(date);
+    let hours = duration.num_hours();
+    Ok(format!(
+        "{} hour(s) and {} minute(s) ago",
+        hours,
+        if hours == 0 {
+            duration.num_minutes()
+        } else {
+            duration.num_minutes() % hours
+        }
+    ))
+}
+
+#[cfg(test)]
+mod tests {
+    use chrono::TimeZone;
+
+    use super::*;
+
+    #[test]
+    fn duration_parses_correctly() {
+        let now = &Utc.ymd(2021, 1, 26).and_hms(19, 38, 0);
+        let duration =
+            human_readable_duration(now, &"2021-01-26T14:00:24+00:00".to_owned()).unwrap();
+        assert!(duration == "5 hour(s) and 2 minute(s) ago")
```

Comment:
Does `assert_eq` not work for this? It has a better error message when the assertion fails in the comparison case.